server/cmd/app.go

```diff
@@ -96,6 +97,49 @@ func (app *Application) listenAndServe() {
 	}
 }
 
+func (app *Application) handleRecommendationsTrends(w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+
+	start, err := time.Parse(time.RFC3339, params.Get("startDateTime"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		doResponse(w, false, "startDateTime must be a valid ISO8601 string")
```

Comment:
API has a different format from admin API (somewhat of a subtle difference), you can add a new method `doApiResponse`. (Don't mind code duplication, it will be refactored later)

API:

```json
{
    "status": "success",
    "data": ...
}
```

admin:

```json
{
    "success": true,
    "data": ...
}
```

Reply:
differentiated the function